app/middleware/get-service-and-gateway-account.middleware.js

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+const _ = require('lodash')
+
+const logger = require('../utils/logger')(__filename)
+const paths = require('../../app/paths')
+const { NotFoundError } = require('../../app/errors')
+const Connector = require('../services/clients/connector.client.js').ConnectorClient
+
+const { keys } = require('@govuk-pay/pay-js-commons').logging
+const connectorClient = new Connector(process.env.CONNECTOR_URL)
+
+async function getGatewayAccountByExternalId (gatewayAccountExternalId, correlationId) {
+  try {
+    const params = {
+      gatewayAccountExternalId: gatewayAccountExternalId,
+      correlationId: correlationId
+    }
+    // TODO: PP-7489 - add additional attributes derived by getAccount middleware
+    return await connectorClient.getAccountByExternalId(params)
+  } catch (err) {
+    const logContext = {}
+    logContext['error'] = err.message
+    logContext['GATEWAY_ACCOUNT_EXTERNAL_ID'] = gatewayAccountExternalId
+    logContext[keys.CORRELATION_ID] = correlationId
+
+    if (err.errorCode === 404) {
+      logger.info('Gateway account not found', logContext)
+      throw new NotFoundError('Gateway account not found')
```

Comment:
I think in some of these cases just logging at the error handling level would be sufficient. We would get the `error.message` logged for free and it also includes things like the correlation ID I believe. 

Streamlining logging here could be done as part of another PR.